python/lsst/pipe/base/task.py

```diff
@@ -127,7 +127,7 @@ def __init__(self, config=None, name=None, parentTask=None, log=None):
                 config = self.ConfigClass()
             self._taskDict = dict()
             loggerName = self._fullName
-            if log is not None:
+            if log is not None and log.getName():
```

Comment:
Yes, an empty string for the root logger.  

Maybe we should improve the error message when we ask for a logger that doesn't make sense...